src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,209 @@
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class CarDaoImpl implements CarDao {
+    @Override
+    public Car create(Car car) {
+        String request = "INSERT INTO cars (model, manufacturer_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createCarStatement = connection
+                         .prepareStatement(request, Statement.RETURN_GENERATED_KEYS)) {
+            createCarStatement.setString(1, car.getModel());
+            createCarStatement.setLong(2, car.getManufacturer().getId());
+            createCarStatement.executeUpdate();
+            ResultSet resultSet = createCarStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Can't create car: " + car, throwable);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String request = "SELECT cars.id AS id, cars.model AS model, "
+                + "manufacturers.id AS manufacturer_id,\n"
+                + "manufacturers.name AS name, manufacturers.country AS country FROM cars \n"
+                + "JOIN manufacturers ON manufacturers.id = cars.manufacturer_id\n"
+                + "WHERE cars.id = ? AND cars.is_deleted = false;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement = connection.prepareStatement(request)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = createCarObject(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get car by id: " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversByCarId(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        List<Car> cars = new ArrayList<>();
+        String request = "SELECT * "
+                + "FROM cars c "
+                + "JOIN manufacturers m "
+                + "   ON c.manufacturer_id = m.id "
+                + "WHERE c.is_deleted = FALSE AND m.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getAllStatement = connection.createStatement()) {
+            ResultSet resultSet = getAllStatement.executeQuery(request);
+            while (resultSet.next()) {
+                cars.add(createCarObject(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars", e);
+        }
+        cars.stream().forEach(car -> car.setDrivers(getDriversByCarId(car.getId())));
```

Comment:
```suggestion
        cars.forEach(car -> car.setDrivers(getDriversByCarId(car.getId())));
```
we can use simple for each without stream